[FEATURE] Weapon/Armor Leveling System (#23)

RedstoneWizard08 opened the pull request:
Closes #23 
WIP DO NOT MERGE

RedstoneWizard08 committed:
[feat] Initial work on weapon leveling & item attributes

naomieow commented on build.gradle:
Please don't include formatting changes in an non-formatting PR.

naomieow commented on src/main/java/com/bibireden/playerex/mixin/ItemStackMixin.java:
```diff
@@ -99,9 +112,10 @@ public <T extends LivingEntity> void preventBreak(int amount, T entity, Consumer
+        ItemStack stack = (ItemStack) (Object) this;
         if (PlayerEXUtil.isBroken(stack) && damage < stack.getDamageValue()) {
             CompoundTag tag = stack.getTag();
+            assert tag != null;
```
We shouldn't throw errors during runtime and this also doesn't belong in this PR

naomieow commented on src/main/java/com/bibireden/playerex/mixin/ItemStackMixin.java:
```diff
@@ -142,7 +156,8 @@ public void preventArmour(EquipmentSlot slot, CallbackInfoReturnable<Multimap<At
     @Unique
     private String playerex$value(double e, Map.Entry<Attribute, AttributeModifier> entry, AttributeModifier modifier) {
-        if (modifier.getOperation() != AttributeModifier.Operation.ADDITION) return ItemStack.ATTRIBUTE_MODIFIER_FORMAT.format(e);
+        if (modifier.getOperation() != AttributeModifier.Operation.ADDITION)
```
Formatting, and we don't use this form of `if` statement - either use a one line as it was previously or surround with curly braces to improve readabiliy.

naomieow commented on src/main/java/com/bibireden/playerex/mixin/ItemStackMixin.java:
Formatting again.

naomieow reviewed (changes requested):
Please do not include formatting changes in a non-formatting PR.

naomieow committed:
Revert some changes
- Revert build.gradle
- Revert gradle.properties
- Remove asserts from ItemStackMixin.java
- Revert some lines in ItemStackMixin.java

naomieow commented:
Please set up GPG/SSH signing for your commits

naomieow committed:
[feat] XP increases on kill
- Weapon XP increases by a value specified in the config when a mob is killed
- Moved XP from an attribute to NBT

RedstoneWizard08 commented:
> Please do not include formatting changes in a non-formatting PR.

Ok, will do.

RedstoneWizard08 commented:
> Please set up GPG/SSH signing for your commits
I'll work with Poke on that, as I don't know how. I've only ever used git over HTTPS.

naomieow commented:
> I'll work with Poke on that, as I don't know how. I've only ever used git over HTTPS.

You can sign your commits with GPG when using HTTPS auth - git just also allows you to sign with SSH as well. I personally use SSH auth and GPG signing. [Here](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) is the GitHub documentation for setting up GPG signing.

naomieow committed:
[feat] Rudimentary item levelling

RedstoneWizard08 commented:
btw @naomieow I can prob work on this again tomorrow, I've been busy with homework until now

naomieow commented:
> btw @naomieow I can prob work on this again tomorrow, I've been busy with homework until now

awesome, am currently working on getting basic armour levelling working right now

naomieow committed:
[feat] Rudimentary armour levelling

naomieow committed:
Merge upstream changes

naomieow committed:
Changes to tooltip colouring

naomieow committed:
[feat] Improve levelling config

naomieow committed:
Merge changes from upstream

naomieow committed:
[fix] Add back accidentally removed translation key

naomieow committed:
[feat] Implement percent damage reduction

naomieow committed:
[feat] Items and weapons in loot tables can generate with a level
- Range they can generate in is different based on dimension
- Non-Vanilla dimensions follow the overworld range

naomieow committed:
[fix] Remove TODO commment

cassie-jfs reviewed (approved):
lgtm
